api.go

```diff
@@ -8,9 +8,16 @@ import (
 	"errors"
 	"fmt"
 	"io"
+	"io/ioutil"
+	"log"
 	"mime/multipart"
 	"net/http"
 	"strconv"
+	"time"
+)
+
+const (
+	timeout = 30 // timeout in seconds for uploads
```

Comment:
You can just make the timeout a duration like this: 
```
const timeout = 30 * time.Second
```

This is more idiomatic and avoids the need to remember/document the that the timeout is in seconds.